gateway/api_loader.go

```diff
@@ -1003,4 +1003,20 @@ func (gw *Gateway) loadApps(specs []*APISpec) {
 
 	mainLog.Info("Initialised API Definitions")
 
+	gwListenPort := gw.GetConfig().ListenPort
+	controlApiIsConfigured := (gw.GetConfig().ControlAPIPort != 0 && gw.GetConfig().ControlAPIPort != gwListenPort) || gw.GetConfig().ControlAPIHostname != ""
+	if gw.allApisAreMTLS() && !gw.GetConfig().Security.ControlAPIUseMutualTLS && !controlApiIsConfigured {
+		mainLog.Warning("All APIs are protected with mTLS, except for the control API. " +
```

Comment:
I wonder if we can simplify and just move this log line to here https://github.com/TykTechnologies/tyk/pull/5018/files#diff-43b9bc3a5fe2dc862ae2eec0d3eb417d369a3ce32b8c3176c784822e17c2c2b1R494

Then you will be able to remove all code in this file and avoid duplication. 

If you afraid that it will spam with logs, you can use sync.Once

Reply:
do you think it could add some delay in the requests? or would it be unnoticeable?

Reply:
also, note that we check if the control api is well configured in this section, basically check if the configs are zero-valued. In cert.go we check if the request will hit the control api (which can be well configured)

Reply:
Agree with @sredxny on the placement of the log, but I agree with @buger on using a `sync.Once` inside the if so we don't spam the `Warn` msg on each reload.

Reply:
Never mind - I'm +1 now on showing it on each reload.

Imagine the following situation:
- All mTLS APIs except one=> no log msg
- We delete the non-mTLS one = all the APIs are mTLS now => show log msg
- Add a non-mTLS one again => no log msg 
- Delete the last added non-mTLS API => no log msg 

^ IMO we should show it again